bindings/java/src/main/native/context.cc

```diff
@@ -9,6 +9,46 @@ extern "C" {
 #include <ucp/core/ucp_mm.h>
 }
 
+/**
+ * Iterates through entries of java's hash map and apply
+ * ucp_config_modify and ucs_global_opts_set_value to each key value pair.
+ */
+static inline void iterate_java_map(JNIEnv *env, ucp_config_t *config,
+                                    jobject *config_map)
+{
+    jclass c_map = env->GetObjectClass(*config_map);
+    jmethodID id_entrySet =
+        env->GetMethodID(c_map, "entrySet", "()Ljava/util/Set;");
+    jclass c_entryset = env->FindClass("java/util/Set");
+    jmethodID id_iterator =
+        env->GetMethodID(c_entryset, "iterator", "()Ljava/util/Iterator;");
+    jclass c_iterator = env->FindClass("java/util/Iterator");
+    jmethodID id_hasNext = env->GetMethodID(c_iterator, "hasNext", "()Z");
+    jmethodID id_next =
+        env->GetMethodID(c_iterator, "next", "()Ljava/lang/Object;");
+    jclass c_entry = env->FindClass("java/util/Map$Entry");
+    jmethodID id_getKey =
+        env->GetMethodID(c_entry, "getKey", "()Ljava/lang/Object;");
+    jmethodID id_getValue =
+        env->GetMethodID(c_entry, "getValue", "()Ljava/lang/Object;");
+    jobject obj_entrySet = env->CallObjectMethod(*config_map, id_entrySet);
+    jobject obj_iterator = env->CallObjectMethod(obj_entrySet, id_iterator);
+
+    while (env->CallBooleanMethod(obj_iterator, id_hasNext)) {
+        jobject entry = env->CallObjectMethod(obj_iterator, id_next);
+        jstring jstrKey = (jstring)env->CallObjectMethod(entry, id_getKey);
+        jstring jstrValue = (jstring)env->CallObjectMethod(entry, id_getValue);
+        const char *strKey = env->GetStringUTFChars(jstrKey, 0);
+        const char *strValue = env->GetStringUTFChars(jstrValue, 0);
+
+        ucp_config_modify(config, strKey, strValue);
+        ucs_global_opts_set_value(strKey, strValue);
```

Comment:
we should probably check for return value: if both have failed print some error that the configuration key,value is invalid

Reply:
Both these methods are void.